src/ui/render.ts

```diff
@@ -112,6 +118,48 @@ export default class UI {
     }
   }
 
+  createFormulaBar(lightSheetContainerDom: HTMLDivElement) {
+    this.formulaBarDom = document.createElement("div");
+    this.formulaBarDom.classList.add("lightsheet_table_formula_bar");
+    lightSheetContainerDom.appendChild(this.formulaBarDom);
+
+    //selected cell display element
+    this.selectedCellDisplay = document.createElement("div");
+    this.selectedCellDisplay.classList.add("selected_cell_display");
+    this.formulaBarDom.appendChild(this.selectedCellDisplay);
+
+    //"fx" label element
+    const fxLabel = document.createElement("div");
+    fxLabel.textContent = "fx";
+    fxLabel.classList.add("fx_label");
+    this.formulaBarDom.appendChild(fxLabel);
+
+    //formula input
+    this.formulaInput = document.createElement("input");
+    this.formulaInput.classList.add("formula_input");
+    this.formulaBarDom.appendChild(this.formulaInput);
+
+    //this.formulaBarDom.style.display = "none";
+    if (this.isReadOnly) {
+      this.formulaBarDom.style.display = "none";
+    } else {
+      this.formulaBarDom.style.display = "flex";
+    }
```

Comment:
when readonly changes, remove or create the formula bar element. Dont just change the display property

Reply:
I'm just wondering why we should remove or create instead of show or hide?

Reply:
Keeping unnessary dom elements in the page is like having unused/unreachable code in backend. I did the same in your toolbar pr.